src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStoreV47.cs

```diff
@@ -0,0 +1,51 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Client;
+
+namespace Microsoft.Health.Dicom.SqlServer.Features.Store;
+
+internal class SqlIndexDataStoreV47 : SqlIndexDataStoreV46
+{
+    public SqlIndexDataStoreV47(SqlConnectionWrapperFactory sqlConnectionWrapperFactory)
+        : base(sqlConnectionWrapperFactory)
+    {
+    }
+
+    public override SchemaVersion Version => SchemaVersion.V47;
+
+
+    public override async Task UpdateFrameDataAsync(int partitionKey, IReadOnlyCollection<long> versions, bool hasFrameMetadata, CancellationToken cancellationToken = default)
+    {
+        var versionRows = versions.Select(i => new WatermarkTableTypeRow(i)).ToList();
```

Comment:
Can we pass the enumerable directly to the `PopulateCommand` method instead of allocating the `List<T>` here?

Reply:
Yes we do need a list since watermark range wont work